### DISCUSSION TITLE
A minimal example towards "Data Art"

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#534] [Q&A]
Artists that use data in their work have been in the radar for many years, from people using self-tracking data to others using publicly available data sources, doing work that is not data viz but that can express "things" through the visual manifestation of data streams.

I know one certain artist, James Schmitz :wink: :wink:, who, often in collaboration with Marina Zurkow, does powerful stuff with data from wind and `<euphemism>`"atmospheric emissions"`</euphemism>`.

So... I'd like for us to have in our docs some minimal example of how to take data from an API, say, temperature, and feed it to a sketch. A long time ago I used to use some pretty easy Processing libraries to interface with weather data I can't even remember the names. 

Is https://open-meteo.com/ the thing I should have a look into nowadays?
This doesn't seem to be real-time, it's prediction I suppose. But anyway...
https://open-meteo.com/en/docs#latitude=-23.5475&longitude=-46.6361&hourly=temperature_2m,rain

https://openweathermap.org/current ?
https://openweathermap.org/api/one-call-3#current

[Some starter code](https://gist.github.com/villares/a9e2fb9f393d4b211765d0572833678a#file-openweathermap-example-py)

What would be a good pattern to consume/update this kind of data, a slowly repeating thread?

**Comment by hx2A** [comment]
It would be great to include some data-related examples in the documentation. What kind of data would be most interesting? All of the data related things I've done have been more long-form works. For *Does The River Flow Both Ways* we do use openweathermap.org and the weather data updates every 15 minutes or so but the weather doesn't actually change that often. Would people have the patience to run a Sketch for hours? Are there other rapidly changing datasets that could be interesting? Besides financial stuff like stock prices, which are boring. Unfortunately with all the data scraping going on, people are more protective of their data.

**Reply by villares** [reply]
AH! For quickly changing stream: public transit data!
I remember a friend did some stuff with São Paulo busses... I'll see if I can find some nice APIs.

**Reply by hx2A** [reply]
Public transit data is a great idea. Here's the data for NYC:

https://new.mta.info/developers
https://api.mta.info/#/

**Reply by hx2A** [reply]
Another data source that people had fun with years ago was Twitter. These days Mastodon has a lot of potential:

https://docs.joinmastodon.org/api/
https://docs.joinmastodon.org/client/intro/

**Reply by Maltiec** [reply]
what about self-generated data, for example mouse activity (moving and clicking)?

**Reply by hx2A** [reply]
> what about self-generated data, for example mouse activity (moving and clicking)?

YES! Love this idea. I did this for a project years ago. Thanks for the reminder! This was a lot of fun and I'd like to do it again.

https://ixora.io/projects/mouse-tracings/
https://ixora.io/galleries/mouse_tracings/

I wrote some Python code to collect keyboard and mouse inputs. I used Processing (this was years ago, before py5) to create these images and set it as the background image for my screen. It would update every 30 seconds or so. When I would start working it would be clear but after hours of work it would be this amazing record of my activity.

I got the idea from https://alllesss.com/ but can't find the actual project they did.
